packages/frontend/src/components/SimplePieChart/PieChartContextMenu.tsx

```diff
@@ -72,20 +86,42 @@ const PieChartContextMenu: FC<PieChartContextMenuProps> = ({
     };
 
     const handleOpenUnderlyingDataModal = () => {
-        // TODO: implement this
-        // openUnderlyingDataModal({
-        //     item,
-        //     value,
-        //     fieldValues: underlyingFieldValues,
-        // });
-        // track({
-        //     name: EventName.VIEW_UNDERLYING_DATA_CLICKED,
-        //     properties: {
-        //         organizationId: user?.data?.organizationUuid,
-        //         userId: user?.data?.userUuid,
-        //         projectId: projectUuid,
-        //     },
-        // });
+        if (!pieChartConfig.selectedMetric) return;
+
+        openUnderlyingDataModal({
+            item: pieChartConfig.selectedMetric,
+            value,
+            fieldValues: {},
+            dashboardFilters,
+            pivotReference: {
```

Comment:
Try using `fieldValues` instead of `pivotReference` I don't think it's needed here